packages/react-components/src/components/LocalVideoTile.tsx

```diff
@@ -54,17 +54,79 @@ export const LocalVideoTile = React.memo(
       localVideoSelectedDescription
     } = props;
 
+    const [localStreamRendererResult, setLocalStreamRendererResult] = React.useState<CreateVideoStreamViewResult>();
+
+    // When localVideoViewOptions change two things may happen:
+    //
+    // 1. Just the scaling mode has changed and we have access to the original LocalStreamRenderView
+    //   - In this case we just need to call updateScalingMode on the view and do not need to recreate the stream view
+    // 2. isMirrored has changed, or scalingMode change and we don't have access to the original LocalStreamRenderView
+    //   - In this case we need to dispose the old view and create a new one
+    //
+    // For scenario 1, we hold onto a ref to the scalingMode that persists across renders. When localVideoViewOptions.scalingMode
+    // differs from the persistent ref, we know to call updateScalingMode. We then subsequently update the ref to the new value for
+    // future renders.
+    //
+    // For scenario 2, we extract the localVideoViewOptions.isMirrored to ensure any change to the isMirrorred prop triggers the useEffect
+    // that recreates the local stream view. We must also here add an extra check for when the scaling mode changes and we do not have
+    // access to the original LocalStreamView (and hence cannot call updateScalingMode). When this happens we must also trigger the useEffect
+    // that recreates the local stream view.
+
+    const scalingModeRef = useRef(localVideoViewOptions?.scalingMode);
+    const newIsMirrored = localVideoViewOptions?.isMirrored;
+    const newScalingMode = localVideoViewOptions?.scalingMode;
+    const hasScalingModeChanged = scalingModeRef.current !== newScalingMode;
+    const updatingScalingModeDirectly = hasScalingModeChanged && !!localStreamRendererResult;
+
+    if (isAvailable && renderElement && newScalingMode && updatingScalingModeDirectly) {
+      localStreamRendererResult && localStreamRendererResult.view.updateScalingMode(newScalingMode);
+      scalingModeRef.current = localVideoViewOptions?.scalingMode;
+    }
+
+    // scalingModeForUseEffect will trigger the useEffect to recreate the local stream view only if the scaling mode has changed and
+    // we cannot call updateScalingMode on the view object directly. Otherwise it will be null to ensure the useEffect does not trigger
+    // when scaling mode changes.
+    const scalingModeForUseEffect = updatingScalingModeDirectly ? null : newScalingMode;
+
     useEffect(() => {
+      // Avoid race condition where onDisposeLocalStreamView is called before onCreateLocalStreamView
+      // and setLocalStreamRendererResult have completed
+      let wasLocalStreamDisposed = false;
```

Comment:
Ugh. Is there *any* way you can extract a (wrapper) component that takes care of just updating the scaling mode?

The logic in this component has been out of hands for a while, and this PR adds more complexity :(

Reply:
Honestly, I almost want to encapsulate all the logic around how we create / update / dispose views outside of React into a class that exposes a simpler API that the react component then uses. React component lifecycle just doesn't seem like a good place to manage the state of the view.

I'd have to give it a spin to really know if such an extraction is feasible. Just a thought: Not everything must be done as part of the React lifecycle.

Reply:
I've moved to a custom hook for now to atleast put the complex logic in one, isolated place, and a place reusable by the remote video stream (will be updating in a seperate PR).
In theory you could have what you're actually wanting, perhaps a manager class stored in a useRef or upgrade the functional component to a class component - but don't have the time to look into this now